MeNext/Playing/PlayController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using MeNext.MusicService;
+
+namespace MeNext
+{
+	/// <summary>
+	/// This class handles actually playing music. It should only be used by the host.
+	/// </summary>
+	public class PlayController
+	{
+		public StatusMessage PreviousMessage { get; set; }
+		public StatusMessage CurrentMessage { get; set; }
+
+		private readonly IMusicService musicService;
+
+		public PlayController(IMusicService musicService)
+		{
+			this.musicService = musicService;
+		}
+
+		/// <summary>
+		/// Updates the actual playing status based on the information in the latest status message
+		/// </summary>
+		/// <param name="message">Message.</param>
+		public void UpdateActualPlaying(StatusMessage message)
+		{
+			PreviousMessage = CurrentMessage;
+			CurrentMessage = message;
+
+			if (CurrentMessage.CurrentSong != null)
+			{
+				var song = musicService.GetSong(CurrentMessage.CurrentSong);
+
+				// Check if a seek has occurred or a new song is playing
+				if (PreviousMessage == null
+					|| PreviousMessage.SongStartTime != CurrentMessage.SongStartTime
+					|| PreviousMessage.CurrentSong != CurrentMessage.CurrentSong)
+				{
+					var positionMs = CurrentMessage.ServerTime - CurrentMessage.SongStartTime;
```

Comment:
The trip time correction needs to be done earlier in the process, _ie_ when the communication itself happens, so we can time how long it takes. The status message is passed thru the message handler so it doesn't know how long the trip took.